tests/helper/helper_oc.go

```diff
@@ -481,3 +481,12 @@ func (oc *OcRunner) GetServices(namespace string) string {
 	output := string(session.Wait().Out.Contents())
 	return output
 }
+
+// VerifyResourceDeleted verifies if the given resource is deleted from cluster
+func (oc *OcRunner) VerifyResourceDeleted(resourceType, resourceName, namespace string) {
+	session := CmdRunner(oc.path, "get", resourceType, "--namespace", namespace)
+	Eventually(session).Should(gexec.Exit(0))
+	output := string(session.Wait().Out.Contents())
```

Comment:
I think this should be `Err.Contents()` because of https://github.com/kubernetes/kubectl/issues/847

Reply:
I am checking for contents, so i guess its ok.

Reply:
`Expect(output).NotTo(ContainSubstring(resourceName))`

Reply:
But I would suggest not using resource names for validation and instead checking if none of the resource type exist because if someone misses a resource name while writing a test, it could go unchecked like in https://github.com/openshift/odo/pull/2732/files#r405976748

Reply:
I feel checking the resource name is less error name, as it is exactly pointing what needs to be checked.

Reply:
yes but it can lead to the above mentioned scenarios

Reply:
I dont think, it is problem, checking `none` lead to entirely missing the error of pvc deletion
https://github.com/openshift/odo/commit/5cb97bcd7dbfbc243e8405857fadea268c4e3db2
I feeld it is better approach.

Reply:
> I dont think, it is problem, checking none lead to entirely missing the error of pvc deletion
5cb97bc

That problem occured due to https://github.com/kubernetes/kubectl/issues/847 and we already have a solution for now. And this approach can lead to https://github.com/openshift/odo/pull/2732/files#r405976748 (highly possible that someone will make this mistake)

Reply:
I missed this, as it has been rebased several time, I am not saying your approach is not right but i have thoroughly tested with this one, there and found it better.

Reply:
> I missed this, as it has been rebased several time, I am not saying your approach is not right but i have thoroughly tested with this one, and hence found it better.

I know you have fixed it and have thoroughly tested it. But my concern is someone else missing something while using this function.

Reply:
and `VerifyResourceDeleted` is checking if that particular resource has been deleted or not, If someday we might not delete user created `pvc` with odo delete, then `No resource found` not work, my point here is in test cases absolute verification is better.

Reply:
hang on. Let me check the usecase first.

Reply:
> then No resource found not work, my point here is in test cases absolute verification is better.

I agree with this point and it will be useful later on but since this test scenario requires checking the existence of all the given resource type, I would like a function to check for the existence of all so we don't miss something by mistake.

Reply:
@adisky https://github.com/openshift/odo/pull/2732#discussion_r406061981